src/subclass/object.rs

```diff
@@ -296,6 +296,7 @@ pub unsafe trait ObjectClassSubclassExt: Sized + 'static {
 unsafe impl ObjectClassSubclassExt for ObjectClass {}
 
 unsafe impl<T: ObjectSubclass> IsSubclassable<T> for ObjectClass {
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::cast_ref_to_mut))]
     fn override_vfuncs(&mut self) {
         unsafe {
             let klass = &mut *(self as *const Self as *mut gobject_ffi::GObjectClass);
```

Comment:
Why do we even cast to a `*const Self`? Just make this a `*mut Self` and it will all be fine :)